### DISCUSSION TITLE
Nocloud Talos refusing to apply config

### DISCUSSION THREAD

**Discussion by HamzaKarh** [siderolabs/talos#9468] [Q&A]
Hi everyone

I have been trying to automate the creation of new Talos VMs on my proxmox server using terraform. I ran into a few issues and it's been a week of trial and error and unfortunately no breakthrough. I'm hoping someone could help here.

I went with nocloud talos iso in order to be able to setup networking automatically and not have to bother fixing everything after OS install. My Talos VMs (one worker and one controller) are sitting on a bridge on my single node proxmox VPS. My hope is to distribute the IP addresses then apply configs and bootstrap the cluster to be able to move on to using K8s. Unfortunately I have been (almost) consistently running into one issue, which is once I apply the config to either the worker or the control plane, they reboot and are back on maintenance. I collected logs and couldn't find anything that would explain this, maybe I'm missing something. 
I also tried applying the config to the created Talos nodes manually, to no avail. I get the same result. I  tried different installers, including the ones provided by the image factory where I get the schematic id and the iso file. 
To compare I tried manually creating a vm based on the normal iso, setting up the network manually through the dashboard, and applying the configs and it works 100% of the time.

Also as an additional note, because the VMs are sitting in a proxmox server to which's network I don't have access directly from my setup, I use ssh tunneling in order to apply configs :
`ssh -L 50000:10.0.1.10:50000 -N -f root@<my_ip> `

**Comment by smira** [comment]
The logs are unreadable, if the issue is failure to install (as you describe), you need to capture the serial console logs in Proxmox.

You might need to create a custom media with extra kernel args `console=tty0 console=ttyS0` via Image Factory, and capture the logs as they go.

**Reply by HamzaKarh** [reply]
It's not on proxmox installation of the ISO, but in the installation that is happening when applying the config :
`[talos] pulling \"factory.talos.dev/installer/376567988ad370138ad8b2698212367b8edcb69b5fd68c80be1f2ec7d603b4ba:v1.8.0\`
Following this it reboots. 
Before that everything seems to be fine.

**Reply by HamzaKarh** [reply]
Also this works without any issues using when it's the metal-amd iso (not an nocloud)

**Reply by smira** [reply]
I can't help much without logs, it's not clear what happens.
